nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -604,22 +600,29 @@ nav_msgs::msg::Path RegulatedPurePursuitController::transformGlobalPlan(
   // We'll discard points on the plan that are outside the local costmap
   nav2_costmap_2d::Costmap2D * costmap = costmap_ros_->getCostmap();
   const double max_costmap_dim = std::max(costmap->getSizeInCellsX(), costmap->getSizeInCellsY());
-  const double max_transform_dist = max_costmap_dim * costmap->getResolution() / 2.0;
+  const double max_costmap_extent = max_costmap_dim * costmap->getResolution() / 2.0;
+  const double max_transform_dist = std::min(max_costmap_extent, dist_to_direction_change);
+
+  // TODO(aposhian): for long paths, this might get computationally expensive to do all the time,
+  // and we should be able to cache the results of this distance accumulation
+  auto closest_pose_upper_bound =
+    nav2_util::geometry_utils::first_element_beyond(
+    global_plan_.poses.begin(), global_plan_.poses.end(), dist_to_direction_change);
```

Comment:
You use `dist_to_direction_change` here, wouldn't this be `max_transform_dist`? What if the distance to direction change is infinite (e.g. there is no direction change).

Reply:
I think that makes sense to use in both locations. That way we only search from point 0 -> costmap edges for the starting point and then we only transform point 0 -> costmap edges for the purposes of control.